process/sync/baseSync.go

```diff
@@ -620,6 +631,8 @@ func (boot *baseBootstrap) requestHeadersFromNonceIfMissing(
 				continue
 			}
 
+			boot.requestedItemsHandler.Add(key)
```

Comment:
If requestedItemsHandler becomes a subcomponent of RequestHandler, these changes might not be necessary.

Reply:
+1

Reply:
This would be refactored after RequestHandler would be refactored and would replace ResolverFinder here